pkg/testing/define/define_flags.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package define
+
+import (
+	"flag"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+type optionalBoolFlag struct {
+	set   bool
+	value bool
+}
+
+func (o *optionalBoolFlag) String() string {
+	if !o.set {
+		return "<not set>"
+	}
+	return strconv.FormatBool(o.value)
+}
+
+func (o *optionalBoolFlag) Set(s string) error {
+	o.set = true
+	o.value = s == "" || s == "true"
+	return nil
+}
+
+func (o *optionalBoolFlag) IsSet() bool {
+	return o.set
+}
+
+func (o *optionalBoolFlag) Value() bool {
+	return o.value
+}
+
+var (
+	DryRun          bool
+	GroupsFilter    []string
+	PlatformsFilter []string
+	SudoFilter      optionalBoolFlag
+
+	groupStringFlag    string
+	platformStringFlag string
+)
+
+func RegisterFlags(prefix string, set *flag.FlagSet) {
+	set.BoolVar(&DryRun, prefix+"dry-run", false, "Forces test in dry-run mode: skips the main test and puts a successful placeholder <TestName>/dry-run if the test would have run")
+	set.StringVar(&groupStringFlag, prefix+"groups", "", "test groups, comma-separated")
+	set.StringVar(&platformStringFlag, prefix+"platforms", "", "test platforms, comma-separated")
+	set.Var(&SudoFilter, prefix+"sudo", "Filter tests by sudo requirements")
+}
+
+func ParseFlags() {
+	GroupsFilter = splitStringToArray(groupStringFlag)
+	PlatformsFilter = splitStringToArray(platformStringFlag)
+}
+
+func splitStringToArray(stringFlag string) []string {
+	if stringFlag == "" {
+		return nil
+	}
+	return strings.Split(stringFlag, ",")
+}
```

Comment:
On the other hand, this looks like it would warrant a `stringArrayFlag` type that would return a slice without need to call any additional functions.

Reply:
Yeah I was kinda writing it quickly and then looked at the stringArraFlag and then forgot to change it :sweat_smile:, I can do it now though

Reply:
done in b05534838d0d1d3c0b1b064540fe0aa277d6fdb9
It had the additional benefit of removing ParseFlags completely (which was a code smell I forgot about)